File: APP-src/lib/models/message.dart
class Message {
  final String id;
  final String type;
  final String title;
  final String content;
  final String? fromUserId;
  final String? fromUserName;
  final String? fromUserAvatar;
  final String? postId;
  final String? commentId;
  final bool isRead;
  final String createdAt;

  Message({
    required this.id,
    required this.type,
    required this.title,
    required this.content,
    this.fromUserId,
    this.fromUserName,
    this.fromUserAvatar,
    this.postId,
    this.commentId,
    this.isRead = false,
    required this.createdAt,
  });

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      id: json['id'] ?? '',
      type: json['type'] ?? 'system',
      title: json['title'] ?? '',
      content: json['content'] ?? '',
      fromUserId: json['from_user_id'],
      fromUserName: json['from_user_name'],
      fromUserAvatar: json['from_user_avatar'],
      postId: json['post_id'],
      commentId: json['comment_id'],
      isRead: json['is_read'] ?? false,
      createdAt: json['created_at'] ?? '',
    );
  }
}

class UnreadCount {
  final int comment;
  final int like;
  final int follow;
  final int at;
  final int system;
  final int total;

  UnreadCount({
    this.comment = 0,
    this.like = 0,
    this.follow = 0,
    this.at = 0,
    this.system = 0,
    this.total = 0,
  });

  factory UnreadCount.fromJson(Map<String, dynamic> json) {
    return UnreadCount(
      comment: json['comment'] ?? 0,
      like: json['like'] ?? 0,
      follow: json['follow'] ?? 0,
      at: json['at'] ?? 0,
      system: json['system'] ?? 0,
      total: json['total'] ?? 0,
    );
  }
}

class MessageSettings {
  final bool commentEnabled;
  final bool likeEnabled;
  final bool followEnabled;
  final bool atEnabled;
  final bool systemEnabled;

  MessageSettings({
    this.commentEnabled = true,
    this.likeEnabled = true,
    this.followEnabled = true,
    this.atEnabled = true,
    this.systemEnabled = true,
  });

  factory MessageSettings.fromJson(Map<String, dynamic> json) {
    return MessageSettings(
      commentEnabled: json['comment_enabled'] ?? true,
      likeEnabled: json['like_enabled'] ?? true,
      followEnabled: json['follow_enabled'] ?? true,
      atEnabled: json['at_enabled'] ?? true,
      systemEnabled: json['system_enabled'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'comment_enabled': commentEnabled,
      'like_enabled': likeEnabled,
      'follow_enabled': followEnabled,
      'at_enabled': atEnabled,
      'system_enabled': systemEnabled,
    };
  }
}

File: Code/src/Pindou.Domain/Entities/Messaging/Message.cs
using SqlSugar;
using Pindou.Domain.Common;

namespace Pindou.Domain.Entities.Messaging;

/// <summary>
/// 消息表
/// </summary>
[SugarTable("messages")]
public class Message : UuidEntity
{
    /// <summary>接收用户ID</summary>
    [SugarColumn(IsNullable = false, Length = 36)]
    public string UserId { get; set; } = string.Empty;

    /// <summary>类型:comment/like/follow/at/system</summary>
    [SugarColumn(Length = 20, IsNullable = false)]
    public string Type { get; set; } = string.Empty;

    /// <summary>标题</summary>
    [SugarColumn(Length = 100, IsNullable = false)]
    public string Title { get; set; } = string.Empty;

    /// <summary>内容</summary>
    [SugarColumn(Length = 500, IsNullable = false)]
    public string Content { get; set; } = string.Empty;

    /// <summary>扩展数据(JSON)</summary>
    [SugarColumn(ColumnDataType = "text", IsNullable = true)]
    public string? Extras { get; set; }

    /// <summary>是否已读</summary>
    [SugarColumn(IsNullable = false, DefaultValue = "0")]
    public bool IsRead { get; set; }
}

/// <summary>
/// 用户消息设置表
/// </summary>
[SugarTable("message_settings")]
public class MessageSetting : UuidEntity
{
    /// <summary>用户ID</summary>
    [SugarColumn(IsNullable = false, Length = 36)]
    public string UserId { get; set; } = string.Empty;

    /// <summary>评论通知</summary>
    [SugarColumn(IsNullable = false, DefaultValue = "1")]
    public bool CommentEnabled { get; set; } = true;

    /// <summary>点赞通知</summary>
    [SugarColumn(IsNullable = false, DefaultValue = "1")]
    public bool LikeEnabled { get; set; } = true;

    /// <summary>关注通知</summary>
    [SugarColumn(IsNullable = false, DefaultValue = "1")]
    public bool FollowEnabled { get; set; } = true;

    /// <summary>@通知</summary>
    [SugarColumn(IsNullable = false, DefaultValue = "1")]
    public bool AtEnabled { get; set; } = true;

    /// <summary>系统通知</summary>
    [SugarColumn(IsNullable = false, DefaultValue = "1")]
    public bool SystemEnabled { get; set; } = true;

    /// <summary>营销通知</summary>
    [SugarColumn(IsNullable = false, DefaultValue = "1")]
    public bool MarketingEnabled { get; set; } = true;

    /// <summary>勿扰模式</summary>
    [SugarColumn(IsNullable = false, DefaultValue = "0")]
    public bool QuietHoursEnabled { get; set; }

    /// <summary>勿扰开始</summary>
    [SugarColumn(Length = 10, IsNullable = false, DefaultValue = "'22:00'")]
    public string QuietHoursStart { get; set; } = "22:00";

    /// <summary>勿扰结束</summary>
    [SugarColumn(Length = 10, IsNullable = false, DefaultValue = "'08:00'")]
    public string QuietHoursEnd { get; set; } = "08:00";
}
